include/_mgp_mock.py

```diff
@@ -0,0 +1,453 @@
+        to_id = to_vertex.id
+
+        self.nx.add_edge(from_id, to_id, key=edge_id, type=edge_type)
+        self._highest_edge_id = edge_id
+
+        return Edge((from_id, to_id, edge_id), self)
+
+    def delete_vertex(self, vertex_id: int):
+        self.nx.remove_node(vertex_id)
+
+    def delete_edge(self, from_vertex_id: int, to_vertex_id: int, edge_id: int):
+        self.nx.remove_edge(from_vertex_id, to_vertex_id, edge_id)
+
+    @property
+    def highest_vertex_id(self) -> int:
+        if self._highest_vertex_id is None:
+            self._highest_vertex_id = max(vertex_id for vertex_id in self.nx.nodes) + 1
+
+        return self._highest_vertex_id
+
+    @property
+    def highest_edge_id(self) -> int:
+        if self._highest_edge_id is None:
+            self._highest_edge_id = max(edge[Graph.I_KEY] for edge in self.nx.edges(keys=True))
+
+        return self._highest_edge_id + 1
+
+
+class Vertex:
+    """Represents a graph vertex."""
+
+    __slots__ = ("_id", "_graph")
+
+    def __init__(self, id: int, graph: Graph) -> None:
+        if not isinstance(id, int):
+            raise TypeError(f"Expected 'int', got '{type(id)}'")
+
+        if not isinstance(graph, Graph):
+            raise TypeError(f"Expected '_mgp_mock.Graph', got '{type(graph)}'")
+
+        if not graph.nx.has_node(id):
+            raise IndexError(f"Unable to find vertex with ID {id}.")
+
+        self._id = id
+        self._graph = graph
+
+    def is_valid(self) -> bool:
+        return self._graph.is_valid()
+
+    def is_deleted(self) -> bool:
+        return not self._graph.nx.has_node(self._id) and self._id <= self._graph.highest_vertex_id
+
+    @property
+    def underlying_graph(self) -> Graph:
+        return self._graph
+
+    def underlying_graph_is_mutable(self) -> bool:
+        return not nx.is_frozen(self._graph.nx)
+
+    @property
+    def labels(self) -> typing.List[int]:
+        return self._graph.nx.nodes[self._id][NX_LABEL_ATTR].split(":")
+
+    def add_label(self, label: str) -> None:
+        if nx.is_frozen(self._graph.nx):
+            raise ImmutableObjectError("Cannot modify immutable object.")
+
+        self._graph.nx.nodes[self._id][NX_LABEL_ATTR] += f":{label}"
+
+    def remove_label(self, label: str) -> None:
+        if nx.is_frozen(self._graph.nx):
+            raise ImmutableObjectError("Cannot modify immutable object.")
+
+        labels = self._graph.nx.nodes[self._id][NX_LABEL_ATTR]
+        if labels.startswith(f"{label}:"):
+            labels = "\n" + labels  # pseudo-string starter
+            self._graph.nx.nodes[self._id][NX_LABEL_ATTR] = labels.replace(f"\n{label}:", "")
+        elif labels.endswith(f":{label}"):
+            labels += "\n"  # pseudo-string terminator
+            self._graph.nx.nodes[self._id][NX_LABEL_ATTR] = labels.replace(f":{label}\n", "")
+        else:
+            self._graph.nx.nodes[self._id][NX_LABEL_ATTR] = labels.replace(f":{label}:", ":")
+
```

Comment:
I would still use here `set()` to store labels, as it is O(1) and much easier to read and maintain code.

Reply:
Keeping this unchanged so that the mock API is compatible out-of-the-box with the [import/export strategies PR](https://github.com/memgraph/gqlalchemy/pull/215).